internal/action/command.go

```diff
@@ -366,9 +366,13 @@ func reloadRuntime(reloadPlugins bool) {
 			continue
 		}
 		if _, ok := parsedSettings[k]; ok {
-			SetGlobalOptionNative(k, parsedSettings[k])
+			err = SetGlobalOptionNative(k, parsedSettings[k])
 		} else {
-			SetGlobalOptionNative(k, defaultSettings[k])
+			err = SetGlobalOptionNative(k, defaultSettings[k])
+		}
+		if err != nil {
+			InfoBar.Error(err)
+			return
```

Comment:
Added the change detection again, but kept the possibility to write a "unchanged" value into the `settings.json`.
Furthermore a moved the evaluation of the global settings into the read function, which catches and corrects the crash above.

Reply:
> Added the change detection again, but kept the possibility to write a "unchanged" value into the `settings.json`.

Why don't you want to *simply* check it here in `reloadRuntime()`, before calling `SetGlobalOptionNative()`?

Reply:
Because for me change detection is something which should be added in a generic way where the value is stored and modified and not into the using component.
Especially when more than one component/path can trigger processing of these values with additional resource costs you like to prevent, then you don't like to add such checks at every position where you use it.

Your expectation might be easier in the current moment. Is it really? Because I don't know. But for future perspective it feels, at least for me, better to place it where the data is hold.

Reply:
Ok, maybe let's do it, but then I think we should at least add a comment explaining why do we write `settings.json` even if the value doesn't change.